app/assets/javascripts/discourse/app/components/composer-editor.js

```diff
@@ -888,6 +894,18 @@ export default Component.extend({
     return element.tagName === "ASIDE" && element.classList.contains("quote");
   },
 
+  _cursorIsOnEmptyLine() {
+    const textArea = this.element.querySelector(".d-editor-input");
+    const selectionStart = textArea.selectionStart;
+    if (selectionStart === 0) {
+      return true;
+    } else if (textArea.value.charAt(selectionStart - 1) === "\n") {
```

Comment:
you could just do:

```
return textArea.value.charAt(selectionStart - 1) === "\n";
```

at the end

Reply:
Or

```javascript
return selectionStart === 0 || textArea.value.charAt(selectionStart - 1) === "\n";
```

instead of the whole `if / else if / else` block 😉

Reply:
Since it's already merged I guess it's better not to spam now with a commit that contains refactoring only?

But I will be writing such conditions shorter since we prefer this style.

Reply:
It's fine to leave it as is. I personally don't mind.